### DISCUSSION TITLE
Connect to Opc ua Server with Basic256Sha256 without certificate

### DISCUSSION THREAD

**Discussion by mcarlino98** [FreeOpcUa/python-opcua#1409] [Q&A] [answered]
Hello!

I'm trying to connect to an opc ua server in anonymous mode with Basic256Sha256 + Sign security settings without certificate.
I don't have the certificate and the server does not need it.
If I try to connect with UaExpert I'm able to connect successfully. 
How I can connect with this settings?

Thanks

**Comment by AndreasHeine** [comment, marked as answer]
> How I can connect with this settings?

Not at all! You need something to Sign with (Its like Signing a Contract without a pen ^^).
UaExpert generates a Certificate while installation and uses it by default!

**Reply by mcarlino98** [reply]
I generate a certificate and now it work, thanks!!!!
